firedrake/interpolation.py

```diff
@@ -664,10 +664,16 @@ def interpolate(self, *function, output=None, transpose=False, **kwargs):
             if transpose:
                 mul = assembled_interpolator.handle.multTranspose
                 V = self.arguments[0].function_space()
+                result = output or firedrake.Function(V)
```

Comment:
This should be a `Cofunction` but it's not enforced anywhere else in interpolation...